tket/src/Utils/MatrixAnalysis.cpp

```diff
@@ -292,173 +292,55 @@ Eigen::VectorXcd apply_qubit_permutation(
   return perm_m * v;
 }
 
+// How close TK2(a,b,c) is from the identity
+double trace_fidelity(double a, double b, double c) {
+  constexpr double g = PI / 2;
+  a *= g;
+  b *= g;
+  c *= g;
+  double trace_sq = 16. * (pow(cos(a) * cos(b) * cos(c), 2) +
+                           pow(sin(a) * sin(b) * sin(c), 2));
+  return (4. + trace_sq) / 20.;
+}
+
 /**
  * returns average fidelity of the decomposition of the information
  * content with nb_cx CNOTS.
  */
-static double get_CX_fidelity(
-    const std::tuple<double, double, double> &k, unsigned nb_cx) {
-  auto fidelity = [](double a, double b, double c) {
-    double trace_sq = 16. * (pow(cos(a) * cos(b) * cos(c), 2) +
-                             pow(sin(a) * sin(b) * sin(c), 2));
-    return (4. + trace_sq) / 20.;
-  };
-
+double get_CX_fidelity(const std::array<double, 3> &k, unsigned nb_cx) {
   TKET_ASSERT(nb_cx < 4);
   auto [a, b, c] = k;
-  constexpr double g = -PI / 2;
-  a *= g;
-  b *= g;
-  c *= g;
 
   // gate fidelity achievable with 0,...,3 cnots
   // this is fully determined by the information content k and is optimal
   // see PhysRevA 71.062331 (2005) for more details on this
   switch (nb_cx) {
     case 0:
-      return fidelity(a, b, c);
+      return trace_fidelity(a, b, c);
     case 1:
-      return fidelity(PI / 4 - a, b, c);
+      return trace_fidelity(0.5 - a, b, c);
```

Comment:
Is this doing the same as before? Before it was multiplying by `-pi/2`, now you are multiplying by `+pi/2`. That doesn't matter for `a,b,c` of course because `sin` etc. are squared, but you are also modifying `a`.

I worry if it should be `0.5+a` instead of `0.5-a`. I assume there are tests somewhere else with specific values to check this? (_With non-symmetric "random" values like `a=0.12345` etc._) We should not rely on algebra alone!

Reply:
Yes, we're safe. But good question. Basically, after the sign swap we project `a` back into the `[0, 0.5]` interval. Now `-0.5` happens to be projected to `0.5`, so this is correct.